From 89fd5470e172431cdd48a0527191f43f65b922e2 Mon Sep 17 00:00:00 2001
From: Hannes Reinecke <hare@suse.de>
Date: Fri, 2 Dec 2016 11:36:13 +0100
Subject: [PATCH 13/29] scsi: hpsa: fallback to use legacy REPORT PHYS command
Git-commit: 2a80d5458a027f6d514987c09f0b51f45c3a5be4
Patch-mainline: v4.10-rc1
References: bsc#1028971

Older SmartArray controllers (eg SmartArray 64xx) do not support the
extended REPORT PHYS command, so fallback to use the legacy version
here.

Signed-off-by: Hannes Reinecke <hare@suse.com>
Acked-by: Don Brace <don.brace@microsemi.com>
Signed-off-by: Martin K. Petersen <martin.petersen@oracle.com>
---
 drivers/scsi/hpsa.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index fed211cbb825..cefa7f404aa2 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -3664,7 +3664,7 @@ static inline int hpsa_scsi_do_report_phys_luns(struct ctlr_info *h,
 
 		/* Copy ReportLUNdata header */
 		memcpy(buf, lbuf, 8);
-		nphys = be32_to_cpu(*((__be32 *)lbuf->LUNListLength)) / 24;
+		nphys = be32_to_cpu(*((__be32 *)lbuf->LUNListLength)) / 8;
 		for (i = 0; i < nphys; i++)
 			memcpy(buf->LUN[i].lunid, lbuf->LUN[i], 8);
 	}
-- 
2.12.0

